parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1165,43 +1169,10 @@ boolean cxxParserParseNextToken(void)
 					// Already have a token to return.
 					// Check again for keywords.
 					
-					// There is a bit of code duplication here but I'd rather
-					// avoid a function call and a switch as the keyword check
-					// must be fast (especially the one above).
-					iCXXKeyword = lookupKeyword(t->pszWord->buffer,g_cxx.eLanguage);
-					if(iCXXKeyword >= 0)
-					{
-						if(
-								(
-									(iCXXKeyword == CXXKeywordFINAL) &&
-									(!g_cxx.bParsingClassStructOrUnionDeclaration)
-								) || (
-									(
-										(iCXXKeyword == CXXKeywordPUBLIC) ||
-										(iCXXKeyword == CXXKeywordPROTECTED) ||
-										(iCXXKeyword == CXXKeywordPRIVATE)
-									) &&
-									(!g_cxx.bEnablePublicProtectedPrivateKeywords)
-								)
-							)
-						{
-							t->eType = CXXTokenTypeIdentifier;
-						} else {
-							t->eType = CXXTokenTypeKeyword;
-							t->eKeyword = (enum CXXKeyword)iCXXKeyword;
-			
-							if(iCXXKeyword == CXXKeyword__ATTRIBUTE__)
-							{
-								// special handling for __attribute__
-								return cxxParserParseNextTokenCondenseAttribute();
-							}
-						}
-					}
-
-					return TRUE;
+					goto check_keyword;
```

Comment:
Yeah resolving loops in the main part would be very tricky, and what do actually do with them?  Better have the user side not re-consider replacements (or if it has to, drop ignore tokens once processed, C-preprocessor-style).